frame/society/src/lib.rs

```diff
@@ -0,0 +1,671 @@
+							value.saturating_sub(tip)
+						}
+					};
+
+					Self::bump_payout(&candidate, maturity, value);
+
+					Some((candidate, total_approvals))
+				} else {
+					Self::suspend_candidate(&candidate, kind);
+					None
+				}
+			}).collect::<Vec<_>>();
+
+			// Reward one of the voters who voted the right way.
+			if !total_slash.is_zero() {
+				if let Some(winner) = pick_item(&mut rng, &rewardees) {
+					// if we can't reward them, not much that can be done.
+					Self::bump_payout(winner, maturity, total_slash);
+				} else {
+					// Move the slashed amount back from payouts account to local treasury.
+					let _ = T::Currency::transfer(&Self::payouts(), &Self::account_id(), total_slash, AllowDeath);
+				}
+			}
+			if !total_payouts.is_zero() {
+				// remove payout from pot and shift needed funds to the payout account.
+				pot = pot.saturating_sub(total_payouts);
+
+				// this should never fail since we ensure we can afford the payouts in a previous
+				// block, but there's not much we can do to recover if it fails anyway.
+				let _ = T::Currency::transfer(&Self::account_id(), &Self::payouts(), total_payouts, AllowDeath);
+			}
+
+			// if at least one candidate was accepted...
+			if !accepted.is_empty() {
+				// select one as primary, randomly chosen from the accepted, weighted by approvals.
+				let primary_point = pick_usize(&mut rng, total_approvals - 1);
+				let primary = accepted.iter().find(|e| e.1 > primary_point)
+					.expect("e.1 of final item == total_approvals; \
+						worst case find will always return that item; qed")
+					.0.clone();
+				let accounts = accepted.into_iter().map(|x| x.0).collect::<Vec<_>>();
+
+				// then write everything back out, signal the changed membership and leave an event.
+				members.sort();
+				<Members<T>>::put(&members);
+				<Head<T>>::put(&primary);
+
+				T::MembershipChanged::change_members_sorted(&accounts, &[], &members);
+				Self::deposit_event(RawEvent::Inducted(primary, accounts));
+			}
+
+			// Bump the pot by at most PeriodSpend, but less if there's not very much left in our
+			// account.
+			let unaccounted = T::Currency::free_balance(&Self::account_id()).saturating_sub(pot);
+			pot += T::PeriodSpend::get().min(unaccounted / 2u8.into());
+
+			<Pot<T>>::put(&pot);
+		}
+
+		// setup the candidates for the new intake
+		let candidates = Self::take_selected(pot);
+		<Candidates<T>>::put(&candidates);
+
+		// initialise skeptics
+		let pick_member = |_| pick_item(&mut rng, &members[..]).expect("exited if members empty; qed");
+		for skeptic in (0..members.len().integer_sqrt()).map(pick_member) {
+			for (_, c, _) in candidates.iter() {
+				<Votes<T>>::insert(c, skeptic, Vote::Skeptic);
+			}
+		}
+	}
+
+	/// Attempt to slash the payout of some member. Return the payout block number most in the
+	/// future along with the total amount deducted.
+	fn slash_payout(who: &T::AccountId, value: BalanceOf<T>) -> (T::BlockNumber, BalanceOf<T>) {
```

Comment:
It is not 100% clear what the returned block number should be here, and what it will be used for. (not currently used in the module)

Right now, it will return the block number of the _first_ payout which still has funds in it. This is not to me "the most in the future".

Reply:
it used to be used and perhaps it'll be used in the future... can be removed if the PR is complete and it's still unused though.

Reply:
removed